src/Scanner.js

```diff
@@ -430,12 +438,10 @@ class DefaultScanner {
         // find the enclosing parent form, and scan it.
         if (realTarget instanceof HTMLInputElement && !realTarget.hasAttribute(ATTR_INPUT_TYPE)) {
             const parentForm = this.getParentForm(realTarget);
-            if (parentForm && parentForm instanceof HTMLFormElement) {
```

Comment:
Wait, why did we remove this conditional here?

Reply:
`getParentForm` can return `HTMLElement` (logical form) too. So this one was probably overprotecting and also in case of reddit.com, there's no real  `HTMLForm` instance.

What maybe we should check here if the returning element is NOT an instance of HTMLInput element (or the same element that went into the function), because that's the only case where we don't want to do anything.

Reply:
I'm about to go on a call and can't double check again right now, but that's exactly what I meant: in this case we were explicitly checking for an `HTMLFormElement` and now we're not. I wonder if we're introducing undesired side effects.

Reply:
I don't think so.

Also `getParentForm` already assumes that forms can be logical only and runs on pages this way. So I am inclined to change it to bailing when, it's an input element, rather than when it's not a form element.